core/src/main/scala/kafka/log/CleanerConfig.scala

```diff
@@ -37,5 +37,6 @@ case class CleanerConfig(numThreads: Int = 1,
                          maxIoBytesPerSecond: Double = Double.MaxValue,
                          backOffMs: Long = 15 * 1000,
                          enableCleaner: Boolean = true,
-                         hashAlgorithm: String = "MD5") {
+                         hashAlgorithm: String = "MD5",
+                         brokerCompressionType: String = "uncompressed") {
 }
```

Comment:
compressionType is a broker/topic(log) config property, not cleaner config property.  This can vary from topic to topic.  We should take compressionType from LogConfigs.